kratos/python/add_factories_to_python.cpp

```diff
@@ -62,6 +67,28 @@ void  AddFactoriesToPython(pybind11::module& m)
      .def("Has",&PreconditionerFactoryType::Has)
     ;
 
+    //////////////////////////////////////////////////////////////
+    //HERE WE REGISTER SOME COMMON METHODS
+    py::class_<FactoryMethods, FactoryMethods::Pointer >(m, "FactoryMethods")
+     .def("Has",&FactoryMethods::Has)
+    ;
+
+    //////////////////////////////////////////////////////////////
+    //HERE THE TOOLS TO REGISTER EXPLICIT BUILDER
+    py::class_<ExplicitBuilderFactoryType, ExplicitBuilderFactoryType::Pointer, FactoryMethods>(m, "ExplicitBuilderFactory")
+     .def( py::init< >() )
+     .def("Create",[](ExplicitBuilderFactoryType& rExplicitBuilderFactory, Kratos::Parameters Settings) {return rExplicitBuilderFactory.Create(Settings);})
+     .def("__str__", PrintObject<ExplicitBuilderFactoryType>)
+    ;
+
+    //////////////////////////////////////////////////////////////
+    //HERE THE TOOLS TO REGISTER EXPLICIT STRATEGIES
+    py::class_<ExplicitStrategyFactoryType, ExplicitStrategyFactoryType::Pointer, FactoryMethods>(m, "ExplicitStrategyFactory")
+     .def( py::init< >() )
+     .def("Create",[](ExplicitStrategyFactoryType& rExplicitStrategyFactory, ModelPart& rModelPart, Kratos::Parameters Settings) {return rExplicitStrategyFactory.Create(rModelPart, Settings);})
```

Comment:
just wondering if we should pass the model instead, so that one can get the modelpart from it

Reply:
if so then this should definitely go to a separate PR as it would change ALL Strategies :sweat_smile:

Reply:
The strategy expects a modelpart, I pass a modelpart